graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -244,6 +246,38 @@ def _process_output_source_directive(schema, current_schema_type, ast,
         return None
 
 
+def _process_tag_directive(context, current_schema_type, location, tag_directive):
+    """Process the tag directive, modifying the context as appropriate.
+
+    Args:
+        context: dict, various per-compilation data (e.g. declared tags, whether the current block
+                 is optional, etc.). May be mutated in-place in this function!
+        current_schema_type: GraphQLType, the schema type at the current location
+        location: Location object representing the current location in the query
+        tag_directive: GraphQL Directive that we want to process
+    """
+    if is_in_fold_scope(context):
+        raise GraphQLCompilationError(u'Tagging values within a @fold vertex field is '
```

Comment:
Let's make sure we have a test for this. I think the module docstring's summary, if followed in the implementation, will ensure that this is handled correctly, but best to prove it with a passing test case.

Reply:
We do.